lib/events.js

```diff
@@ -85,36 +85,50 @@ EventEmitter.prototype.emit = function emit(type) {
   if (this.domain && this !== process)
     this.domain.enter();
 
+  len = arguments.length;
   if (util.isFunction(handler)) {
-    switch (arguments.length) {
+    switch (len) {
       // fast cases
       case 1:
         handler.call(this);
         break;
       case 2:
-        handler.call(this, arguments[1]);
+        handler.call(this, arg1);
         break;
       case 3:
-        handler.call(this, arguments[1], arguments[2]);
+        handler.call(this, arg1, arg2);
         break;
       // slower
       default:
-        len = arguments.length;
         args = new Array(len - 1);
         for (i = 1; i < len; i++)
           args[i - 1] = arguments[i];
         handler.apply(this, args);
     }
-  } else if (util.isObject(handler)) {
-    len = arguments.length;
-    args = new Array(len - 1);
-    for (i = 1; i < len; i++)
-      args[i - 1] = arguments[i];
-
+  } else if (util.isArray(handler)) {
```

Comment:
I just noted during rebasing that the conflicts are due to the code having changed to move away from `util` operations. So I'll do this in general.